basyx.components/basyx.components.updater/basyx.components.updater.camel-aas/src/main/java/basyx/components/updater/aas/AASProducer.java

```diff
@@ -34,9 +34,23 @@ public AASProducer(AASEndpoint endpoint) {
 
 	@Override
 	public void process(Exchange exchange) throws Exception {
-		LOG.info("Got msg for property: " + exchange.getMessage().getBody(String.class));
-    	connectedProperty.setValue(exchange.getMessage().getBody(String.class));
+    	String messageBody = exchange.getMessage().getBody(String.class);
+		String fixedMessageBody = "";
+    	fixedMessageBody = fixMessage(messageBody);
+
+		LOG.info("Got msg for property: " + fixedMessageBody);
+    	connectedProperty.setValue(fixedMessageBody);
     }
+
+    String fixMessage(String messageBody) {
+		String fixedMessageBody = "";
+		if (messageBody.contains("\"") && messageBody != null && messageBody.isEmpty() == false) {
```

Comment:
There are multiple issues with this line:
- messageBody != null is checked too late, messageBody.contains() is executed before the null check
- messageBody.contains() will return false for an empty string, thus "messageBody.isEmpty == false" is not needed here.
- messageBody.contains("\\"") also returns true for string _test"123_. Would it be possible to use messageBody.startsWith("\\"") && messageBody.endsWith("\\"") instead?

Reply:
Thanks for pointing that out!
I will change those lines.